test/others/libcriu/test_feature_check.c

```diff
@@ -0,0 +1,65 @@
+#include "criu.h"
+#include <fcntl.h>
+#include <stdio.h>
+#include <errno.h>
+#include <stdlib.h>
+#include <signal.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+
+#include "lib.h"
+
+int main(int argc, char **argv)
+{
+	int ret;
+	char *env;
+	int mem_track = 0;
+	int lazy_pages = 0;
+	int pidfd_store = 0;
+	struct criu_feature_check features = {
+		.mem_track = true,
+		.lazy_pages = true,
+		.pidfd_store = true,
+	};
+
+	printf("--- Start feature check ---\n");
+	criu_init_opts();
+	criu_set_service_binary(argv[1]);
+
+	env = getenv("CRIU_FEATURE_MEM_TRACK");
+	if (env) {
+		mem_track = atoi(env);
+	}
+	env = getenv("CRIU_FEATURE_LAZY_PAGES");
+	if (env) {
+		lazy_pages = atoi(env);
+	}
+	env = getenv("CRIU_FEATURE_PIDFD_STORE");
+	if (env) {
+		pidfd_store = atoi(env);
+	}
+
+	ret = criu_feature_check(&features, sizeof(features) + 1);
+	printf("   `- passing too large structure to libcriu should return -1: %d\n", ret);
+	if (ret != -1)
+		return -1;
+
+	ret = criu_feature_check(&features, sizeof(features));
+	if (ret < 0) {
+		what_err_ret_mean(ret);
+		return ret;
+	}
+
+	printf("   `- mem_track  : %d - expected : %d\n", features.mem_track, mem_track);
+	if (features.mem_track && mem_track != 1)
```

Comment:
Check is strange... Why not `features.mem_track != mem_track`? I believe your check does not detect the case where mem_track is true and features.mem_track is false.

Reply:
fixed